compiler/modules/comm.diagnostic/diagnostic.bal

```diff
@@ -110,6 +113,92 @@ public function messageFormat(Template t) returns string {
     return string:concat(...strs);
 }
 
+public type DiagnosticError error<Diagnostic>;
+
+public type DiagnosticPrinter object {
+    public function addDiagnostic(DiagnosticError diagnostic);
```

Comment:
Call the method `print` and `flush`. Call the type `Printer` (others will use it with `d:` prefix). Move into a separate file `print.bal`.

Reply:
Fixed. I think flush is a reserved keyword. Renamed it to `flushPrinter`